ignite/src/test/java/org/hibernate/ogm/datastore/ignite/utils/IgniteTestConfigurationBuilder.java

```diff
@@ -0,0 +1,416 @@
+import org.apache.ignite.cache.QueryIndexType;
+import org.apache.ignite.cache.affinity.rendezvous.RendezvousAffinityFunction;
+import org.apache.ignite.configuration.BinaryConfiguration;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.IgnitionEx;
+import org.apache.ignite.internal.binary.BinaryMarshaller;
+import org.apache.ignite.logger.slf4j.Slf4jLogger;
+import org.hibernate.ogm.datastore.ignite.IgniteConfigurationBuilder;
+
+/**
+ * Ignite cache configuration for tests
+ *
+ * @author Victor Kadachigov
+ */
+public class IgniteTestConfigurationBuilder implements IgniteConfigurationBuilder {
+
+	@Override
+	public IgniteConfiguration build() {
+		//disable check for new versions
+		System.setProperty( IgniteSystemProperties.IGNITE_UPDATE_NOTIFIER, Boolean.FALSE.toString() );
+		
+		IgniteConfiguration config = null;
+
+		try {
+			//config = loadFromResource( "ignite-config.xml" );
+			config = createConfig();
+		}
+		catch (Exception ex) {
+			throw new RuntimeException( ex );
+		}
+
+		return config;
+	}
+
+	private IgniteConfiguration loadFromResource(String resourceName) throws IgniteCheckedException {
+		URL url = getClass().getClassLoader().getResource( resourceName );
+		return IgnitionEx.loadConfiguration( url ).get1();
+	}
+
+	private IgniteConfiguration createConfig() {
+		IgniteConfiguration config = new IgniteConfiguration();
+
+		config.setGridName( "OgmTestGrid" );
+		config.setLocalHost( "127.0.0.1" );
+		config.setPeerClassLoadingEnabled( true );
+		config.setClientMode( false );
+		config.setMarshaller( new BinaryMarshaller() ) ;
+		BinaryConfiguration binaryConfiguration = new BinaryConfiguration();
+		binaryConfiguration.setCompactFooter( false );		// it is necessary only for embedded collections (@ElementCollection)
+		config.setBinaryConfiguration( binaryConfiguration );
+		config.setGridLogger( new Slf4jLogger() );
+		config.setPublicThreadPoolSize( 2 );
+
+		List<CacheConfiguration> cacheConfig = new ArrayList<>();
+
+// EmbeddableIdTest
+		cacheConfig.add( createCacheConfig( "SingleBoardComputer" ).withForceQueryEntity().build() );
+// SequenceIdGeneratorTest
+		cacheConfig.add( simpleCacheConfig( "Song", Long.class ) );
+		cacheConfig.add( simpleCacheConfig( "Actor", Long.class ) );
+// JPAResourceLocalTest
+		cacheConfig.add( simpleCacheConfig( "Poem" ) );
+// ManyToManyTest
+		cacheConfig.add(
+				createCacheConfig( "Car" )
+					.withKeyType( "CarId" )
+					.build()
+		);
+		cacheConfig.add(
+				createCacheConfig( "Tire" )
+					.withKeyType( "TireId" )
+					.build()
+		);
+		cacheConfig.add(
+				createCacheConfig( "Car_Tire" )
+					.appendIndex( "cars_carId_maker", String.class )
+					.appendIndex( "cars_carId_model", String.class )
+					.appendIndex( "tires_tireId_maker", String.class )
+					.appendIndex( "tires_tireId_model", String.class )
+					.build()
+		);
```

Comment:
>  Isn't it possible to move this logic in the IgniteCacheInitializer?

No. I don't think so. This class is for real work not for tests.
And why? What's wrong with IgniteTestConfigurationBuilder?